client/client.go

```diff
@@ -465,6 +465,54 @@ func (c *client) parseRegionResponse(res *pdpb.GetRegionResponse) *Region {
 	return r
 }
 
+func (c *client) waitForRegion(ctx context.Context, key []byte) (*pdpb.GetRegionResponse, error) {
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
The old `ctx` should also be used to detect the `<-ctx.Done()`, otherwise the upper timeout will be meaningless.

Reply:
please use a different variable name for the new `ctx` to improve the code readability.